MPDZ.py

```diff
@@ -0,0 +1,471 @@
+    return:
+        - list: transformed sequences with separators
+    """
+    res = []
+    for seq in seqs:
+        threel_form = ""
+        for aa in seq:
+            threel_form = threel_form + threel[aa] + sep
+        if sep:
+            threel_form = threel_form[:-1]
+        res.append(threel_form)
+    return res
+
+
+coperations = {
+    "length": length_info,
+    "percentage": count_percentage_aa,
+    "pattern": find_pattern,
+    "3Letter_name": rename_three_letter_name,
+    "DNA_code": get_protein_gene,
+}
+
+
+retrnaslation_dict = {
+    "F": "TTC",
+    "f": "ttc",
+    "L": "TTA",
+    "l": "tta",
+    "S": "TCG",
+    "s": "tcg",
+    "Y": "TAC",
+    "y": "tac",
+    "C": "TGC",
+    "c": "tgc",
+    "W": "TGG",
+    "w": "tgg",
+    "P": "CCC",
+    "p": "ccc",
+    "H": "CAT",
+    "h": "cat",
+    "Q": "GAA",
+    "q": "gaa",
+    "R": "CGA",
+    "r": "cga",
+    "I": "ATT",
+    "i": "att",
+    "M": "ATG",
+    "m": "atg",
+    "T": "ACC",
+    "t": "acc",
+    "N": "AAT",
+    "n": "aat",
+    "K": "AAA",
+    "k": "aaa",
+    "V": "GTT",
+    "v": "gtt",
+    "A": "GCA",
+    "a": "gca",
+    "D": "GAT",
+    "d": "gca",
+    "E": "GAG",
+    "e": "gag",
+    "G": "GGG",
+    "g": "ggg",
+}
+
+
+threel = {
+    "A": "ALA",
+    "R": "ARG",
+    "N": "ASN",
+    "D": "ASP",
+    "V": "VAL",
+    "H": "HIS",
+    "G": "GLY",
+    "Q": "GLN",
+    "E": "GLU",
+    "I": "ILE",
+    "L": "LEU",
+    "K": "LYS",
+    "M": "MET",
+    "P": "PRO",
+    "S": "SER",
+    "Y": "TYR",
+    "T": "THR",
+    "W": "TRP",
+    "F": "PHE",
+    "C": "CYS",
+    "a": "ala",
+    "r": "arg",
+    "n": "asn",
+    "d": "asp",
+    "v": "val",
+    "h": "his",
+    "g": "gly",
+    "q": "gln",
+    "e": "glu",
+    "i": "ile",
+    "l": "leu",
+    "k": "lys",
+    "m": "met",
+    "p": "pro",
+    "s": "ser",
+    "y": "tyr",
+    "t": "thr",
+    "w": "trp",
+    "f": "phe",
+    "c": "cys",
+}
+
+
+def protein_tool(*proteins, options=None):
+    proteins = list(proteins)
+
+    operations = {
+        "compare": compare,
+        "length": length_info,
+        "percentage": count_percentage_aa,
+        "pattern": find_pattern,
+        "3Letter_name": rename_three_letter_name,
+        "DNA_code": get_protein_gene,
+    }
+
+    if options == "compare":
+        result = operations[options](proteins[:-2], proteins[-2], proteins[-1])
+        return result
+    elif options == "3Letter_name":
+        result = operations[options](proteins[:-1], proteins[-1])
+        return result
+    elif options == "length" or options == "percentage" or options == "DNA_code":
+        result = []
+        for protein in proteins:
+            res = operations[options](protein)
+            result.append(res)
+        return result
+    else:
+        raise ValueError("Incorrect options input, please try again")
+
+
+from Bio import SeqIO
+#from Bio.SeqUtils import GC
+
+class FastQFilter:

```

Comment:
По условиям задания, можно было обойтись написанием функции, использующей классы из модуля `Bio`.

К тому же, новые классы нужны, когда мы хотим ввести интерфейс для взаимодействия с ранее не использовавшимися типами данных. Т.е. классы создаются под тип данных, а не под операции с ними; для операций вводятся методы класса. Поэтому логичнее было бы назвать класс просто `FastQ`, а не `FastQFilter`:

```suggestion
class FastQ:
```